### ISSUE TITLE
AttributeError: Either 'OrderDetail' object has no attribute 'passed_obj' or...

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#2422]
_Imported from trac ticket 2422.
Created by **sal**
Opened in trac: Jul 25, 2013, 10:50:05 AM
Last modified in trac: Jul 25, 2013, 11:57:13 AM_
***
_by sal:_
_From Sal:_
I get this error when trying to flag line 12 and 13 on SO 0712331AC69 to fill from this store

```
   merged_obj = self.modelClass.save(passed_obj, controller_validate)
 File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 881 in save
   obj.run_rules()
 File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 22379 in run_rules
   self._update_all_autoadjustments()
 File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 22403 in _update_all_autoadjustments
   o._update_autoadjustments()
 File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 7120 in _update_autoadjustments
   adj._save_autoadjustment()
 File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 7738 in _save_autoadjustment
   l._shared_set_defaults()
 File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 9577 in _shared_set_defaults
   if 'purchaseordered' not in self.passed_obj:
 File '/home/rarch/trunk/src/appserver/pylotengine/model/objects.py', line 209 in __getattr__
   raise AttributeError("Either '{0}' object has no attribute '{1}' or evaluating '{1}' caused an AttributeError".format(self.__class__.__name__, name))
 AttributeError: Either 'OrderDetail' object has no attribute 'passed_obj' or evaluating 'passed_obj' caused an AttributeError
```

***
* kbower attached [untitled-part.png](https://trac.retailarchitects.com/trac/attachment/ticket/2422/untitled-part.png) on Jul 25, 2013, 11:55:52 AM

**Comment by kentbower** [comment]
_**kbower** trac comment on Jul 25, 2013, 10:50:35 AM:_

This appears to happen fairly often...

**Comment by kentbower** [comment]
_**kbower** trac comment on Jul 25, 2013, 11:55:52 AM:_

Sal,
  This was occurring when a user needed to add or remove coverage from an
FCP123 line that was already closed.  For example, the user adds another
line item to the open A ticket and then wants that line covered by FCP123.
Since FCP123 is already closed, Loft creates an automatic MDB for the extra
$30 if the extra item brought the total of covered items to the next price
tier level.  (It might also have been occurring for other reasons Loft
needed to add an automatic adjustment.)

This will be fixed in the next release.

Thanks for finding it (I noticed it occurred on TRAIN order 0624305CI29,
which was very helpful in figuring out the problem!)


On Fri, Jul 19, 2013 at 10:33 AM, Sal <sal@ivansmith.com> wrote:

[[Image(untitled-part.png)]]

**Comment by kentbower** [comment]
_**kbower** trac comment on Jul 25, 2013, 11:57:13 AM:_

(Changeset [[4375]](../commit/8350cf15021fc277fd30cccab580c88512bd550c))
* fixes #2422, AttributeError: "Either 'OrderDetail' object has no attribute 'passed_obj' or..." when autoadjustment needed #all

**Comment by kentbower** [comment]
_**kbower** trac status change on Jul 25, 2013, 11:57:13 AM:_ **closed**
